src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -90,6 +90,91 @@ public Utf8JsonReader(in ReadOnlySequence<byte> jsonData, bool isFinalBlock, Jso
             }
         }
 
+        /// <summary>
+        /// Constructs a new <see cref="Utf8JsonReader"/> instance.
+        /// </summary>
+        /// <param name="jsonData">The ReadOnlySequence&lt;byte&gt; containing the UTF-8 encoded JSON text to process.</param>
+        /// <param name="options">Defines the customized behavior of the <see cref="Utf8JsonReader"/>
+        /// that is different from the JSON RFC (for example how to handle comments or maximum depth allowed when reading).
+        /// By default, the <see cref="Utf8JsonReader"/> follows the JSON RFC strictly (i.e. comments within the JSON are invalid) and reads up to a maximum depth of 64.</param>
+        /// <remarks>
+        ///   <para>
+        ///     Since this type is a ref struct, it is a stack-only type and all the limitations of ref structs apply to it.
+        ///   </para>
+        ///   <para>
+        ///     This assumes that the entire JSON payload is passed in directly (equivalent to <see cref="IsFinalBlock"/> = true)
+        ///   </para>
+        /// </remarks>
+        public Utf8JsonReader(in ReadOnlySequence<byte> jsonData, JsonReaderOptions options = default)
+        {
```

Comment:
Can't we implement this much more simply, with something like:
```C#
public Utf8JsonReader(in ReadOnlySequence<byte> jsonData, JsonReaderOptions options = default) :
    this(jsonData, isFinalBlock: true, new JsonReaderState(options))
{
}
```
?  It'd be really good to avoid adding all of this duplication.

Reply:
Though equivalent, doing it this way does slows the ctor down by ~20%:

**Before:**
``` ini

BenchmarkDotNet=v0.11.3.1003-nightly, OS=Windows 10.0.18362
Intel Core i7-6700 CPU 3.40GHz (Skylake), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=3.0.100-preview7-012499
  [Host]     : .NET Core 3.0.0-preview7-27825-11 (CoreCLR 4.700.19.32501, CoreFX 4.700.19.32514), 64bit RyuJIT
  Job-MWKMSV : .NET Core ? (CoreCLR 4.700.19.32401, CoreFX 4.700.19.32601), 64bit RyuJIT

Runtime=Core  Toolchain=CoreRun  

```
|           Method |     Mean |     Error |    StdDev |   Median |      Min |      Max | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|----------------- |---------:|----------:|----------:|---------:|---------:|---------:|------------:|------------:|------------:|--------------------:|
|             Ctor | 74.20 ns | 1.5188 ns | 2.5376 ns | 72.88 ns | 70.86 ns | 79.78 ns |           - |           - |           - |                   - |
| CtorMultiSegment | 74.42 ns | 0.8864 ns | 0.7858 ns | 74.56 ns | 72.52 ns | 75.55 ns |           - |           - |           - |                   - |


**After:**

|           Method |     Mean |    Error |   StdDev |   Median |      Min |      Max | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|----------------- |---------:|---------:|---------:|---------:|---------:|---------:|------------:|------------:|------------:|--------------------:|
|             Ctor | 88.95 ns | 1.262 ns | 1.119 ns | 89.24 ns | 86.76 ns | 91.31 ns |           - |           - |           - |                   - |
| CtorMultiSegment | 90.89 ns | 1.347 ns | 1.260 ns | 90.71 ns | 88.58 ns | 93.50 ns |           - |           - |           - |                   - |


52 ns vs 65 ns for the span overload of the ctor

Reply:
And what about doing something real?

I'd rather pay the extra 10ns and not duplicate this mass of code. It also seems like something that the JIT should be able to improve.